File: apps/web/src/types/strategyAnalysis.ts
export type StrategyName = "wholesale" | "wholetail" | "flip" | "rental_hold";
export type StrategyResult = {
  strategy_result_id?: string; strategy: StrategyName; viable: boolean; rank?: number; score: number; confidence: number;
  acquisition_price_used?: number; value_basis?: string; value_low?: number; value_high?: number;
  rehab_cost_low?: number; rehab_cost_high?: number; holding_costs?: number; closing_costs?: number; selling_costs?: number;
  projected_revenue_low?: number; projected_revenue_high?: number; projected_profit_low?: number; projected_profit_high?: number;
  projected_return_percentage_low?: number; projected_return_percentage_high?: number; monthly_cash_flow?: number; annual_cash_flow?: number; cap_rate?: number;
  major_risks: string[]; strengths: string[]; missing_inputs: string[]; formula_notes: string[];
};
export type StrategyAnalysis = {
  strategy_analysis_id: string; property_id: string; recommended_strategy?: StrategyName; recommendation_confidence: number;
  analysis_status: string; analyzed_at: string; assumptions_version: string; missing_inputs: string[]; limitations: string[];
  recommended_next_action: string; input_snapshot: Record<string, unknown>; results: StrategyResult[];
};

File: services/api/app/schemas/strategy_analysis.py
from typing import Literal

from pydantic import BaseModel, Field


StrategyName = Literal["wholesale", "wholetail", "flip", "rental_hold"]


class StrategyResult(BaseModel):
    strategy: StrategyName
    viable: bool
    rank: int | None = None
    score: float = Field(ge=0, le=100)
    confidence: float = Field(ge=0, le=1)
    acquisition_price_used: float | None = None
    value_basis: str | None = None
    value_low: float | None = None
    value_high: float | None = None
    rehab_cost_low: float | None = None
    rehab_cost_high: float | None = None
    holding_costs: float | None = None
    closing_costs: float | None = None
    selling_costs: float | None = None
    projected_revenue_low: float | None = None
    projected_revenue_high: float | None = None
    projected_profit_low: float | None = None
    projected_profit_high: float | None = None
    projected_return_percentage_low: float | None = None
    projected_return_percentage_high: float | None = None
    monthly_cash_flow: float | None = None
    annual_cash_flow: float | None = None
    cap_rate: float | None = None
    major_risks: list[str]
    strengths: list[str]
    missing_inputs: list[str]
    formula_notes: list[str]


class StrategyAnalysis(BaseModel):
    strategy_analysis_id: str | None = None
    property_id: str
    recommended_strategy: StrategyName | None = None
    recommendation_confidence: float = Field(ge=0, le=1)
    analysis_status: str
    analyzed_at: str
    assumptions_version: str
    missing_inputs: list[str]
    limitations: list[str]
    recommended_next_action: str
    input_snapshot: dict
    results: list[StrategyResult]
